Source/WebCore/Modules/speech/cocoa/WebSpeechRecognizerTask.mm

```diff
@@ -220,12 +220,14 @@ - (void)speechRecognitionTask:(SFSpeechRecognitionTask *)task didHypothesizeTran
 - (void)speechRecognitionTask:(SFSpeechRecognitionTask *)task didFinishRecognition:(SFSpeechRecognitionResult *)recognitionResult
 {
     ASSERT(isMainThread());
+
+    if (task.state == SFSpeechRecognitionTaskStateCanceling || (!_doMultipleRecognitions && task.state == SFSpeechRecognitionTaskStateCompleted))
```

Comment:
if a user starts recognition and stops immediately (before this callback is invoked), will the end event be dispatched?
Should we [self sendEndIfNeeded]; if `(!_doMultipleRecognitions && task.state == SFSpeechRecognitionTaskStateCompleted))`?

Reply:
>Should we [self sendEndIfNeeded]; if (!_doMultipleRecognitions && task.state == SFSpeechRecognitionTaskStateCompleted))?

We can't do that because continuous recognition sits in SFSpeechRecognitionTaskStateCompleted as it returns more results (which seems odd, but is what happens).

>If a user starts recognition and stops immediately (before this callback is invoked), will the end event be dispatched?

I will check!